apps/spruce/src/pages/version/TestAnalysis/index.tsx

```diff
@@ -0,0 +1,215 @@
+import { useState } from "react";
+import { useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import Button from "@leafygreen-ui/button";
+import { Combobox, ComboboxOption } from "@leafygreen-ui/combobox";
+import { BasicEmptyState } from "@leafygreen-ui/empty-state";
+import { palette } from "@leafygreen-ui/palette";
+import { SearchInput } from "@leafygreen-ui/search-input";
+import { ListSkeleton } from "@leafygreen-ui/skeleton-loader";
+import { H3, Label, Body, H3Props } from "@leafygreen-ui/typography";
+import pluralize from "pluralize";
+import { failedTaskStatuses, taskStatusToCopy } from "constants/task";
+import { size } from "constants/tokens";
+import {
+  TestAnalysisQuery,
+  TestAnalysisQueryVariables,
+} from "gql/generated/types";
+import { TEST_ANALYSIS } from "gql/queries";
+import FailedTestGroup from "./FailedTestGroup";
+import {
+  countTotalTests,
+  filterGroupedTests,
+  getAllBuildVariants,
+  getAllTaskStatuses,
+  groupTestsByName,
+} from "./utils";
+
+const { green } = palette;
+interface TestAnalysisProps {
+  versionId: string;
+}
+const TestAnalysis: React.FC<TestAnalysisProps> = ({ versionId }) => {
+  const [selectedTaskStatuses, setSelectedTaskStatuses] = useState<string[]>(
+    [],
+  );
+  const [selectedBuildVariants, setSelectedBuildVariants] = useState<string[]>(
+    [],
+  );
+  const [searchValue, setSearchValue] = useState<string>("");
+  const { data, loading } = useQuery<
```

Comment:
Loading this tab on prod with lots of failed tests chokes the page. Not sure exactly if the data is inaccurate or where the error originates. We should either fix the error, or implement pagination or some other sort of limit to avoid this situation. I produced the performance slowdown this on the tab for version `66fc06e0c3df2c0007b55c22`. `66fbc5745743660007ed2bd8` is another version that is able to be rendered but has so many tests present that it suffers from a subpar UX imo.

Reply:
😮 Great catch this was actually an error in grouping the tasks which caused a test to repeat itself thousands of times if it occurred in multiple tasks. I fixed that logic and made sure it was covered in a test.